lib/basket/backend_adapter/redis_backend.rb

```diff
@@ -44,8 +42,12 @@ def read(queue)
 
       private
 
+      def serialize_data(data)
+        JSON.generate(data)
```

Comment:
I'd love to bake is some sort of validation here that checks if the data is truly "JSONifiable" since the default `class` will return `to_s`:

```
class Foo; end;

JSON.generate(data)
=> "\"#<Hi:0x0000000>\""
```